File: Matury CKE/2022 Czerwiec/PROGRAMOWANIE/python/zad4_3.py
# numbers=[i.strip() for i in open('../../DANE/przyklad.txt').readlines()]
numbers=[i.strip() for i in open('../../DANE/liczby.txt').readlines()]


def is_prime(n):
    for i in range(2, n):
        if n%i==0:
            return False
    return True


primes_with_prime_reflections=[i for i in numbers if is_prime(int(i)) and is_prime(int(i[::-1]))]
print(f'4.3. {" ".join(primes_with_prime_reflections)}')

File: Matury CKE/2022 Czerwiec/Programowanie/C++/zad4_3.cpp
#include <iostream>
#include <string>
#include <fstream>
#include <vector>
#include <algorithm>
#include <cmath>

int reflection(std::string str)
{
    std::reverse(str.begin(), str.end());
    return std::stoi(str);
}

bool isFirst(int num)
{
    for (int i = 2; i <= sqrt(num); i++) {
      if (num % i == 0) {
        return false;
      }
    }
    return true;
}

int main()
{
    std::cout << "4.3. " << std::endl;

    // std::ifstream dane("C:/Matury-Informatyka/Matury CKE/2022 Czerwiec/DANE/przyklad.txt");
    std::ifstream dane("C:/Matury-Informatyka/Matury CKE/2022 Czerwiec/DANE/liczby.txt");

    std::vector<std::string> numsList;
    std::vector<int> areBothFirstList;

    std::string num;
    while(dane >> num){
        numsList.push_back(num);
        int numNow = std::stoi(num);
        int refNum = reflection(num);

        if(isFirst(numNow) && isFirst(refNum))
        {
            std::cout << numNow << std::endl;
        }
    }

    return 0;
}
